### ISSUE TITLE
The documented example 'nix-shell can now be used as a #!-interpreter' doesn't work

### ISSUE THREAD

**Issue by deepfire** [NixOS/nix#846]
On latest `closure-size` the example at http://nixos.org/nix/manual/#ssec-relnotes-1.9 of how to write a "#!" interpreter breaks:

```
[deepfire@andromedae:~/src]$ cat > fooscript.hs
#! /usr/bin/env nix-shell
#! nix-shell -i runghc -p haskellPackages.ghc haskellPackages.HTTP

import Network.HTTP

main = do
  resp <- Network.HTTP.simpleHTTP (getRequest "http://nixos.org/")
  body <- getResponseBody resp
  print (take 100 body)

[deepfire@andromedae:~/src]$ chmod +x fooscript.hs

[deepfire@andromedae:~/src]$ ./fooscript.hs

fooscript.hs:4:8:
    Could not find module ‘Network.HTTP’
    Use -v to see a list of the files searched for.
```

**Comment by deepfire** [comment]
@peti, perhaps you might help..

**Comment by peti** [comment]
That code works only in `release-14.12` and earlier. Ever since the switch to Haskell NG, that example has been broken. The proper way to do it is:

~~~~
#! /usr/bin/env nix-shell
#! nix-shell -i runghc -p "haskellPackages.ghcWithPackages (ps: with ps; [HTTP])"

import Network.HTTP

main = do
  resp <- Network.HTTP.simpleHTTP (getRequest "http://nixos.org/")
  body <- getResponseBody resp
  print (take 100 body)
~~~~

**Comment by deepfire** [comment]
The example still fails for me on recent `closure-size`:
```
[deepfire@andromedae:~]$ src/fooscript.hs
build input haskellPackages.ghcWithPackages) does not exist
```
Unfortunately I can't cheaply check how `master` fares..

**Comment by peti** [comment]
The problem might be that the version of Nix available in that branch is too old for that script to work. It does work in `master`, which has `nix-shell` version 1.11.2.

**Comment by deepfire** [comment]
```
[nix-shell:~/src]$ nix-env --version
nix-env (Nix) 1.10
```
Maybe it is..

**Comment by deepfire** [comment]
Hmm, oddly `closure-size` has had `master` merged into it about a week ago..

@vcunat, perhaps you can provide some insight?

**Comment by copumpkin** [comment]
@deepfire have you tried upgrading your `nix`?

**Comment by vcunat** [comment]
closure-size contains nix-1.11.2. There were no changes in its version. The version you see depends on the way you update nix. It's also possible that your `$NIX_PATH` is pointing somewhere you don't expect, e.g. some old channel you don't update.

**Comment by deepfire** [comment]
@peti, your example now works for me on recent `master`.

The Nix manual still refers to 14.12, though: http://nixos.org/nix/manual/#use-as-a-interpreter

..so I guess it needs to be updated before we can close this?

**Comment by peti** [comment]
Technically, the example is correct because that code works in 14.12. Not sure what to do. Would you like to suggest a patch / PR?

**Comment by sevanspowell** [comment]
I ran into this again today on NixOS 18.09, can this be fixed now?
